kwanghwi/myapp/app.js

```diff
@@ -37,20 +39,55 @@ app.get("/ping", (req, res) => {
 });
 
 app.post("/user/signup", async (req, res) => {
-  const { userId, password, email, profileImage } = req.body;
+  const { userId, name, password, email, profileImage } = req.body;
+  const saltRounds = 12;
+
+  const makeHash = async (password, saltRounds) => {
+    return await bcrypt.hash(password, saltRounds);
+  };
+
+  const hashedPassword = await makeHash(password, saltRounds);
+
   await appDataSource.query(
     `INSERT INTO users(
       user_id,
+      name,
       password,
       email,
-      profileImage
-    ) VALUES (?, ?, ?, ?);
+      profile_image
+    ) VALUES (?, ?, ?, ?, ?);
     `,
-    [userId, password, email, profileImage]
+    [userId, name, hashedPassword, email, profileImage]
   );
   return res.status(201).json({ message: "signup success!" });
 });
 
+app.post("/user/signin", async (req, res) => {
+  const { userId, password } = req.body;
+
+  const [selectQuery] = await appDataSource.query(
+    `SELECT
+      user_id,
+      password
+    FROM users
+    WHERE user_id = ?
+    `,
+    [userId]
+  );
+
+  const checkHash = async (password, hashedPassword) => {
+    return await bcrypt.compare(password, hashedPassword);
+  };
+
+  const result = await checkHash(password, selectQuery.password);
```

Comment:
내부 로직이 복잡하거나 가독성이 떨어지는 것이 아니기 때문에 굳이 function으로 만들지 않아도 될 것 같습니다-! 
또한 bcrypt.compare()를 통해 return 되는 결과값은 boolean이기 때문에 이를 통해 비밀번호가 매칭되지 않았을 경우 핸들링이 필요할 것 같네요.
```suggestion
  const match = await bcrypt.compare(password, user.password)
  
  if (!match) {
  . . . 
  }
```